pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,220 @@ def func(df):
+                    key=key,
+                    axis=self.axis,
+                    level=self.level,
+                    sort=self.sort,
+                    mutated=self.mutated,
+                    dropna=dropna,
+                )
+                groupings += list(grouper.groupings)
+
+            # Take the size of the overall columns
+            result = df.groupby(
+                groupings,
+                as_index=self.as_index,
+                sort=self.sort,
+                observed=self.observed,
+                dropna=dropna,
+            ).size()
+
+            # Change the nameof the size result
+            if self.as_index:
+                result.name = RESULT_NAME
+            else:
+                result.rename({"size": RESULT_NAME}, axis=1, inplace=True)
+
+            if normalize:
+                # Normalize the results by dividing by the original group sizes
+                indexed_group_size = df.groupby(
+                    main_grouper, sort=self.sort, observed=self.observed, dropna=dropna
+                ).size()
+                if self.as_index:
+                    if index_grouping:
+                        # The common index needs a common name
+                        indexed_group_size.index.set_names("Group", inplace=True)
+                        result.index.set_names("Group", level=0, inplace=True)
+                    # Use indexed group size series
+                    result /= indexed_group_size
+                    if index_grouping:
+                        result.index.set_names(None, level=0, inplace=True)
+                else:
+                    # Make indexed key group size series
+                    indexed_group_size.name = "group_size"
+                    if index_grouping:
+                        # Get the column name of the added groupby index column
+                        index_column_name = result.columns[0]
+                        indexed_group_size.index.set_names(
+                            index_column_name, inplace=True
+                        )
+                        left_on = index_column_name
+                    else:
+                        left_on = keys
+                    if not index_grouping and len(keys) == 1:
+                        # Compose with single key group size series
+                        group_size = indexed_group_size[result[keys[0]]]
+                    else:
+                        # Merge multiple key group size series
+                        merged = result.merge(
+                            indexed_group_size,
+                            how="left",
+                            left_on=left_on,
+                            right_index=True,
+                        )
+                        group_size = merged["group_size"]
+
+                    result[RESULT_NAME] /= group_size.values
+
+            if sort:
+                # Sort the values and then resort by the main grouping
+                if self.as_index:
```

Comment:
Similar to the above - if we pass as_index=True and then adjust after, do we only need the if clause here.

Reply:
As above